### PR TITLE
Feature/8528-payment

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:* /bcgov/entity/8528

*Description of changes:*

- Added services to consume getlinkedrouitngslips for a particular routing slip
- Moved paymentinfo to a child component

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/ViewRoutingSlip/LinkedRoutingSlipsPaymentInfo.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<template>
```
Comment (karthik-aot) [inline src/components/ViewRoutingSlip/LinkedRoutingSlipsPaymentInfo.vue:1]: This is WIP component. Would be adding code for displaying children payment information in here

File: src/composables/ViewRoutingSlip/useLinkedRoutingSlipsPaymentInfo.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Payment } from '@/models/Payment'
+import { PaymentMethods } from '@/util/constants'
+import { RoutingSlip } from '@/models/RoutingSlip'
+import { createNamespacedHelpers } from 'vuex-composition-helpers'
+
+const routingSlipModule = createNamespacedHelpers('routingSlip') // specific module name
+const { useState } = routingSlipModule
+
+// Composable function to inject Props, options and values to PaymentDetails component
+export default function useLinkedRoutingSlipsPaymentInfo () {
```
Comment (karthik-aot) [inline src/composables/ViewRoutingSlip/useLinkedRoutingSlipsPaymentInfo.ts:10]: This is WIP component. Would be adding code for displaying children payment information in here

File: src/store/modules/routingSlip.ts [hunk 3]
```diff
@@ -253,4 +271,41 @@ export default class RoutingSlipModule extends VuexModule {
...
+      const context: any = this.context
+      if (response && response.data && response.status === 200) {
+        /* The response to getLinkedRoutingSlips consists of following values:
+        1. children: non empy, parent: empty: which implies that the routingslip is a parent slip
+        2. children: empty, parent: non empty: which implies that the slip is a child slip
+        3. no content 204 resposne: non linked routingslip where we need to reset the states
+        */
+        if (response.data?.children) {
+          const linkedRoutingSlips: LinkedRoutingSlips = {
+            children: response.data?.children,
```
Comment (shabeeb-aot) [inline src/store/modules/routingSlip.ts:288]: can we remove if else and refactor?

File: src/store/modules/routingSlip.ts [hunk 4]
```diff
@@ -253,4 +271,41 @@ export default class RoutingSlipModule extends VuexModule {
...
+            parent: {}
+          }
+          context.commit('setLinkedRoutingSlips', linkedRoutingSlips)
+        } else if (response.data?.parent) {
+          const linkedRoutingSlips: LinkedRoutingSlips = {
+            children: [],
+            parent: response.data?.parent
+          }
+          context.commit('setLinkedRoutingSlips', linkedRoutingSlips)
```
Comment (shabeeb-aot) [inline src/store/modules/routingSlip.ts:297]: commit to store after all calculation

**Review by shabeeb-aot — APPROVED** [review]

**Review by saravanpa-aot — APPROVED** [review]

File: src/store/modules/routingSlip.ts (hunk 3 above)
Comment (karthik-aot) [reply src/store/modules/routingSlip.ts]: yeah . Thanks, we have totally removed if :-P

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**